pkg/gateway/mirror/service/mirror.go

```diff
@@ -37,21 +37,25 @@ import (
 type Mirror interface {
 	Start(ctx context.Context) (<-chan error, error)
 	Connect(ctx context.Context, targets ...*payload.Mirror_Target) error
+	Disconnect(ctx context.Context, targets ...*payload.Mirror_Target) error
+	IsConnected(ctx context.Context, addr string) bool
 	MirrorTargets() ([]*payload.Mirror_Target, error)
+	RangeAllMirrorAddr(f func(addr string, _ any) bool)
+	SelfMirrorAddrs() []string
 }
 
 type mirr struct {
-	addrl         sync.Map // List of all connected addresses
-	selfMirrAddrs []string // Address of self mirror gateway
-	selfMirrAddrl sync.Map // List of self Mirror gateway addresses
-	gwAddrs       []string // Address of Vald Gateway (LB gateway)
-	gwAddrl       sync.Map // List of Vald Gateway addresses
-	gateway       Gateway
+	addrl         sync.Map                 // List of all connected addresses
+	selfMirrTgts  []*payload.Mirror_Target // Targets of self mirror gateway
+	selfMirrAddrs []string                 // Address of self mirror gateway
+	selfMirrAddrl sync.Map                 // List of self Mirror gateway addresses
```

Comment:
Sorry, could you explain why you use multiple variables which seem to have nearly equal values?

Reply:
All of the information is the same, but the data structure is different. 
If only slice variables are used, a loop will run every time when checking for the existence of an element, which may increase the calculation cost. Therefore, I define variables that represent the same data in a different data structure. However, the slice variable (`selfMirrTgts `) and the sync.Map variable (`selfMirrAddrl `) can satisfy the requirement, I will remove the `selfMirrAddrs ` variable.